src/core/atomicdex/pages/qt.settings.page.cpp

```diff
@@ -606,12 +585,83 @@ namespace atomic_dex
             }
         };
 
-        functor_remove(std::move(wallet_cfg_path));
+        if (fs::exists(wallet_cfg_path))
+        {
+            nlohmann::json wallet_config_json_data;
+            std::unordered_set<std::string> active_list;
+            QFile          fs;
+            fs.setFileName(std_path_to_qstring(wallet_cfg_path));
+            fs.open(QIODevice::ReadOnly | QIODevice::Text);
+
+            //! Read Contents
+            wallet_config_json_data = nlohmann::json::parse(QString(fs.readAll()).toStdString());
+            fs.close();
+
+            //! Get the active coins
+            for (auto&& [key, value]: wallet_config_json_data.items())
+            {
+                if (value["active"]) { active_list.insert(key); }
+            }
+
+            // remove old coins file
+            functor_remove(std::move(wallet_cfg_path));
+
+            //! Copy default coins file
+            const auto  cfg_path = ag::core::assets_real_path() / "config";
+            std::string filename = std::string(atomic_dex::get_raw_version()) + "-coins.json";
+            fs::copy(cfg_path / filename, wallet_cfg_path);
+
+            //! Open coins file
+            fs.setFileName(std_path_to_qstring(wallet_cfg_path));
+            fs.open(QIODevice::ReadOnly | QIODevice::Text);
+
+            //! Read Contents
+            wallet_config_json_data = nlohmann::json::parse(QString(fs.readAll()).toStdString());
+            fs.close();
+
+            //! set active coins again
+            for (auto&& key: active_list)
+            {
+                if (active_list.contains(key))
```

Comment:
If you iterate through the active list, the key is already present so the if condition is useless here.

Please also rename `active_list` to `active_coins_registry`